test/endpoints/baz/baz_metrics_test.go

```diff
@@ -146,7 +147,77 @@ func TestCallMetrics(t *testing.T) {
 		assert.Equal(t, expectedEndpoitTags[tag.GetTagName()], tag.GetTagValue())
 	}
 
-	tchannelOutboundSuccessMetric := metrics[3]
+	expectedJaegerSpanMetricName := "test-gateway.production.all-workers.jaeger.spans"
```

Comment:
It would be really good to be able to write tests on the spans emitted themselves instead of just the metrics jaeger emits.